src/Chapter2.hs

```diff
@@ -500,7 +523,16 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+
+i :: Int
+i = error "not implemented"
+isThird42 :: [Int] -> Bool
+isThird42 [] = False
+isThird42 x = if length x < 3 then False else check42 x
+    where
+      check42 x = case (x !! 2) of
+        42 -> True
+        _ -> False
```

Comment:
You can implement this check with pattern matching as well. It would be cheaper as we won't need to take `length` first.
As you have `length`, there is another problem with such function: you can't stream on them. And this is a problem for lazy language.
With pattern patching implementation, you would be able to use this function even on infinite lists instantly!

```haskell
isThird42 (_:_:42:_) = True
isThird _ = False
```